projects/igniteui-angular/src/lib/grids/row-drag.directive.ts

```diff
@@ -0,0 +1,129 @@
+import { Directive, Input, OnDestroy, NgModule, Output, EventEmitter } from '@angular/core';
+import { IgxDragDirective, IgxDragCustomEventDetails } from '../directives/dragdrop/dragdrop.directive';
+import { IRowDragEndEventArgs, IRowDragStartEventArgs } from './grid-base.component';
+import { KEYS } from '../core/utils';
+import { fromEvent, Subscription } from 'rxjs';
+import { IgxRowComponent, IgxGridBaseComponent, IGridDataBindable } from './grid';
+
+const ghostBackgrounClass = 'igx-grid__tr--ghost';
+const draggedRowClass = 'igx-grid__tr--drag';
+
+/**
+ * @hidden
+ */
+@Directive({
+    selector: '[igxRowDrag]'
+})
+export class IgxRowDragDirective extends IgxDragDirective implements OnDestroy {
+    private _row: IgxRowComponent<IgxGridBaseComponent & IGridDataBindable>;
+    private subscription$: Subscription;
+
+    @Input('igxRowDrag')
+    set data(val) {
+        this._row = val;
+    }
+
+    get row() {
+        return this._row;
+    }
+
+    public onPointerDown(event) {
+        event.preventDefault();
+        super.onPointerDown(event);
+
+        const args: IRowDragStartEventArgs = {
+            source: this.row
+        };
+
+        this.row.grid.onRowDragStart.emit(args);
+        this.row.dragging = true;
+        this.row.grid.rowDragging = true;
+
+        this.subscription$ = fromEvent(this.row.grid.document.defaultView, 'keydown').subscribe((ev: KeyboardEvent) => {
+            if (ev.key === KEYS.ESCAPE || ev.key === KEYS.ESCAPE_IE) {
+                this.onPointerUp(event);
+            }
+        });
+    }
+
+    public onPointerMove(event) {
+        super.onPointerMove(event);
+
+        if (this._dragGhost && this._lastDropArea != null) {
```

Comment:
Why check twice for _dragGhost. Could be like this:
        if (this._dragGhost) {
            const dragIndicator = this._dragGhost.getElementsByClassName('igx-grid__tr--drag-indicator')[0];
            dragIndicator.getElementsByTagName('igx-icon')[0].innerHTML = this._lastDropArea ? 'add' : 'block';
        }